app/src/main/java/org/oppia/android/app/application/OppiaApplication.kt

```diff
@@ -22,7 +23,7 @@ class OppiaApplication :
       .build()
   }
 
-  override fun createActivityComponent(activity: AppCompatActivity): ActivityComponent {
+  override fun createActivityComponent(activity: AppCompatActivity): ActivityComponentImpl {
```

Comment:
The contract comes from ActivityComponentFactory which must return ActivityComponent (in order for the dependencies to build correctly in Gradle, and conceptually an impl shouldn't be returned). This is just overriding the return type to be more specific.

Updated this to return ActivityComponent instead since it doesn't need to actually be more specific in order to build.